src/overlay025/poketch_system.c

```diff
@@ -0,0 +1,790 @@
+static void PoketchSystem_LoadApp(PoketchSystem *poketchSys, int param1);
+static void PoketchSystem_UnloadApp(PoketchSystem *poketchSys);
+static void ov25_022541D8(PoketchSystem *poketchSys, u32 param1);
+static BOOL ov25_022541FC(PoketchSystem *poketchSys);
+static void ov25_0225420C(PoketchSystem *poketchSys);
+static BOOL ov25_02254228(PoketchSystem *poketchSys);
+static BOOL ov25_02254284(PoketchSystem *poketchSys);
+static void ov25_022542C8(PoketchSystem *poketchSys);
+static void ov25_022542D4(PoketchSystem *poketchSys);
+static void ov25_022542E4(u32 param0, u32 param1, u32 param2, void * param3);
+static BOOL ov25_022543EC(UnkStruct_ov25_02254560 * param0, u32 param1);
+static inline BOOL inline_ov25_0225446C(u32 param0, u32 param1);
+struct PoketchSystem * sub_0203D158(void);
+
+// the order of this array determines the app order in the poketch.
+static const struct {
+    int appID;
+    FSOverlayID overlayID;
+} sAppOverlayIDs[] = {
+    {POKETCH_APPID_DIGITALWATCH,        FS_OVERLAY_ID(overlay26)},
+    {POKETCH_APPID_STOPWATCH,           FS_OVERLAY_ID(overlay27)},
+    {POKETCH_APPID_CALCULATOR,          FS_OVERLAY_ID(overlay28)},
+    {POKETCH_APPID_MEMOPAD,             FS_OVERLAY_ID(overlay29)},
+    {POKETCH_APPID_ANALOGWATCH,         FS_OVERLAY_ID(overlay30)},
+    {POKETCH_APPID_DOTART,              FS_OVERLAY_ID(overlay31)},
+    {POKETCH_APPID_PARTYSTATUS,         FS_OVERLAY_ID(overlay32)},
+    {POKETCH_APPID_FRIENDSHIPCHECKER,   FS_OVERLAY_ID(overlay33)},
+    {POKETCH_APPID_DOWSINGMACHINE,      FS_OVERLAY_ID(overlay34)},
+    {POKETCH_APPID_COUNTER,             FS_OVERLAY_ID(overlay35)},
+    {POKETCH_APPID_PEDOMETER,           FS_OVERLAY_ID(overlay36)},
+    {POKETCH_APPID_DAYCARECHECKER,      FS_OVERLAY_ID(overlay40)},
+    {POKETCH_APPID_ROULETTE,            FS_OVERLAY_ID(overlay41)},
+    {POKETCH_APPID_COINTOSS,            FS_OVERLAY_ID(overlay42)},
+    {POKETCH_APPID_MOVETESTER,          FS_OVERLAY_ID(overlay43)},
+    {POKETCH_APPID_MATCHUPCHECKER,      FS_OVERLAY_ID(overlay44)},
+    {POKETCH_APPID_ALARMCLOCK,          FS_OVERLAY_ID(overlay45)},
+    {POKETCH_APPID_KITCHENTIMER,        FS_OVERLAY_ID(overlay46)},
+    {POKETCH_APPID_MARKINGMAP,          FS_OVERLAY_ID(overlay47)},
+    {POKETCH_APPID_BERRYSEARCHER,       FS_OVERLAY_ID(overlay48)},
+    {POKETCH_APPID_COLORCHANGER,        FS_OVERLAY_ID(overlay49)},
+    {POKETCH_APPID_CALENDAR,            FS_OVERLAY_ID(overlay50)},
+    {POKETCH_APPID_LINKSEARCHER,        FS_OVERLAY_ID(overlay52)},
+    {POKETCH_APPID_RADARCHAINCOUNTER,   FS_OVERLAY_ID(overlay53)},
+    {POKETCH_APPID_POKEMONHISTORY,      FS_OVERLAY_ID(overlay54)}
+};
+
+static PoketchSystem * ov25_02253CE0 (void)
+{
+    return sub_0203D158();
+}
+
+
+void PoketchSystem_Create(FieldSystem *fieldSys, PoketchSystem **poketchSys, SaveData *saveData, BGL *param3, NNSG2dOamManagerInstance *param4)
+{
+    PoketchSystem *new_system;
+
+    Heap_Create(3, 7, 49152);
+    Heap_Create(3, 8, 49152);
```

Comment:
From `include/constants/heap.h`, heap 3 is `HEAP_ID_APPLICATION`.

Are heaps 7 and 8 initialized anywhere else? Can we define constants for them now?

Can we also define a constant for `49152` in `constants/heap.h`?